MTM-49761 added Microservice Security troubleshooting (#792)
SoftwareAG/c8y-docs: content/microservice-sdk/java-bundle/troubleshooting.md

4. Try running a Docker command again.

Also refer to [Docker Engine > Installation per distro > Optional post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/) in the Docker documentation.

> ##### Customizing Microservice Security configuration

Reviewer: ```suggestion
##### Customizing microservice security configuration
```

> Starting from version 10.17 there was a change in internal Microservice Security configuration following

Reviewer: ```suggestion
Starting from version 10.17 there was a change in the internal microservice security configuration following
```

> deprecation of `WebSecurityConfigurerAdapter` by Spring Security. Microservice SDK is not using direct

Reviewer: ```suggestion
the deprecation of `WebSecurityConfigurerAdapter` by Spring Security. The Microservice SDK does no longer use a direct
```

> declaration of `SecurityFilterChain` bean in its internal configuration any more. At the same time Spring Security

Reviewer: ```suggestion
declaration of the `SecurityFilterChain` bean in its internal configuration. At the same time, Spring Security
```

> only allows one of those configuration approaches in a single application. This means that if the old,

Reviewer: ```suggestion
only allows one of these configuration approaches in a single application. This means that if the old,
```

> adapter based method was used in your code before, you will have to migrate to the new, direct filters

Reviewer: ```suggestion
adapter-based method has been used in your code before, you will have to migrate to the new, direct filters
```

> declaration for application to start. Please refer to [Spring Security documentation](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/config.html#_stop_using_websecurityconfigureradapter) for more details.

Reviewer 2: Do we have something in our examples? If so I would add in last sentence:
`Please refer to [Spring Security documentation] and examples in [cumulocity-examples] respository for more details.

Author: No, we do not have any custom security configuration in examples.

Reviewer 2: :-(  maybe we should have at least one...

> declaration for application to start. Please refer to [Spring Security documentation](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/config.html#_stop_using_websecurityconfigureradapter) for more details.

Reviewer: ```suggestion
declaration for applications to start. Refer to the [Spring Security documentation](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/config.html#_stop_using_websecurityconfigureradapter) for more details.
```
